faiss_vector_posting.go

```diff
@@ -295,22 +295,32 @@ func (sb *SegmentBase) InterpretVectorIndex(field string) (
 			return rv, nil
 		}
 
-		scores, ids, err := vecIndex.Search(qVector, k)
+		var vectorIDsToExclude []int64
+		// iterate through the vector doc ID map and if the doc ID is one to be
+		// deleted, add it to the list
+		for vecID, docID := range vecDocIDMap {
+			if except != nil && except.Contains(docID) {
+				vectorIDsToExclude = append(vectorIDsToExclude, vecID)
+			}
+		}
+
+		var scores []float32
+		var ids []int64
+		var err error
+		scores, ids, err = vecIndex.SearchWithoutIDs(qVector, k, vectorIDsToExclude)
```

Comment:
i think `scores, ids, err := vecIndex.SearchWithoutIDs(qVector, k, vectorIDsToExclude)` should suffice over here right?